tests/st/calicoctl/test_crud.py

```diff
@@ -35,745 +35,1091 @@ class TestPool(TestBase):
-                                       'action': 'deny',
-                                       'destination': {
-                                           'notNets': ['10.3.0.0/16'],
-                                           'notPorts': ['110:1050'],
-                                           'notSelector': "type=='apples'",
-                                           'notTag': "bananas",
-                                           'nets': ['10.2.0.0/16'],
-                                           'ports': ['100:200'],
-                                           'selector': "type=='application'",
-                                           'tag': 'alphatag'},
-                                       'icmp': {'type': 10, 'code': 6},
-                                       'protocol': 'tcp',
-                                       'source': {
-                                           'notNets': ['10.1.0.0/16'],
-                                           'notPorts': [1050],
-                                           'notSelector': "type=='database'",
-                                           'notTag': 'bartag',
-                                           'nets': ['10.0.0.0/16'],
-                                           'ports': [1234, '10:20'],
-                                           'selector': "type=='application'",
-                                           'tag': "production"}}],
-                      }}),
-        ("profile2", {'apiVersion': 'projectcalico.org/v2',
-                      'kind': 'Profile',
-                      'metadata': {
-                          'name': 'profile2',
-                      },
-                      'spec': {
-                          'egress': [{'action': 'allow',
-                                      'destination': {},
-                                      'source': {}}],
-                          'ingress': [{'ipVersion': 6,
-                                       'action': 'deny',
-                                       'destination': {},
-                                       'source': {}}],
-                      }}),
-    ]
+        # Attempt to replace using the old revision (rev0).  This will fail.
+        rc = calicoctl("replace", data=rev0)
+        rc.assert_error(text=ERROR_CONFLICT)
 
-    @parameterized.expand(testdata)
-    def test_create_from_file_yaml(self, name, data):
-        self._check_data_save_load(data)
-        res_type = data['kind']
-        logger.debug("Testing %s" % res_type)
-        # Write out the files to load later
-        self.writeyaml('/tmp/%s-1.yaml' % res_type, data)
+        # Replace using the original data, but with the new resource version.
+        rev0['metadata']['resourceVersion'] = rev1['metadata']['resourceVersion']
+        rc = calicoctl("replace", data=rev0)
+        rc.assert_no_error()
 
-        calicoctl("create", "/tmp/%s-1.yaml" % res_type)
+        # Delete the resource without using a resource version.
```

Comment:
how does this `rev0` not have a `resourceVersion`? Didn't you set it in line 135?

Reply:
because we are only using the name to do the delete.

Reply:
i've tweaked the comment to spell this out.